identity_presentation/src/jvmTest/kotlin/net/thechance/mena/identity/presentation/screen/addresses/AddressesScreenViewModelTest.kt

```diff
@@ -0,0 +1,162 @@
+package net.thechance.mena.identity.presentation.screen.addresses
+
+import app.cash.turbine.test
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.mockk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import net.thechance.mena.identity.domain.entity.Address
+import net.thechance.mena.identity.domain.entity.AddressType
+import net.thechance.mena.identity.domain.repository.AddressRepository
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+import kotlin.uuid.ExperimentalUuidApi
+import kotlin.uuid.Uuid
+
+@OptIn(ExperimentalCoroutinesApi::class, ExperimentalUuidApi::class)
+class AddressesScreenViewModelTest {
+
+    private val testDispatcher = StandardTestDispatcher()
+    private val addressRepository: AddressRepository = mockk(relaxed = true)
+    private lateinit var viewModel: AddressesScreenViewModel
+
+    @BeforeTest
+    fun setUp() {
+        Dispatchers.setMain(testDispatcher)
+        viewModel = AddressesScreenViewModel(addressRepository, testDispatcher)
+    }
+
+    @AfterTest
+    fun tearDown() {
+        Dispatchers.resetMain()
+    }
+
+    @Test
+    fun `should fetch addresses on init and update state on success`() = runTest {
```

Comment:
Adding function names would be great 
```suggestion
    fun `init() should fetch addresses on init and update state on success`() = runTest {
```